File: clientlibs/csharp/src/identifiers/Enums.cs
namespace Enums
{
  public enum IMetricMetaData
  {
    CONTINUOUS,
    CATEGORICAL
  }

  public enum PAYLOAD_TYPE
  {
    STRING,
    JSON,
    CSV
  }

  public enum SUPPORTED_CALIPER_PROFILES
  {
    GRADING_PROFILE
  }

  public enum SUPPORTED_CALIPER_EVENTS
  {
    GRADE_EVENT
  }

  public enum REQUEST_TYPES
  {
    GET,
    POST,
    PATCH
  }

  public enum MARKED_DECISION_POINT_STATUS
  {
    CONDITION_APPLIED,
    CONDITION_FAILED_TO_APPLY,
    NO_CONDITION_ASSIGNED
  }

  public enum EXPERIMENT_TYPE
  {
    SIMPLE,
    FACTORIAL
  }
}

File: clientlibs/python/src/upgrade_client_lib/types/enums.py
"""Enums for the UpGrade client library."""

from enum import Enum


class MarkedDecisionPointStatus(str, Enum):
    CONDITION_APPLIED = "condition applied"
    CONDITION_FAILED_TO_APPLY = "condition not applied"
    NO_CONDITION_ASSIGNED = "no condition assigned"


class ExperimentType(str, Enum):
    SIMPLE = "Simple"
    FACTORIAL = "Factorial"


class PayloadType(str, Enum):
    STRING = "string"
    JSON = "json"
    CSV = "csv"


class BinaryRewardValue(str, Enum):
    SUCCESS = "SUCCESS"
    FAILURE = "FAILURE"
